js/views/SurveyDetailsPage.js

```diff
@@ -19,145 +19,61 @@ import SurveyListItem from '../components/SurveyListItem'
 import Loading from '../components/Loading'
 import Button from '../components/Button'
 
+import { InvitationStatus, markInvitationStatus } from '../api/Invitations'
+
 const SurveyDetailsPage = React.createClass ({
   propTypes: {
     survey: React.PropTypes.object.isRequired, // Realm.io Object
+    formCount: React.PropTypes.number.isRequired,
+    questionCount: React.PropTypes.number.isRequired,
   },
 
   getInitialState() {
-    let cachedSurvey = this.props.survey
     return {
-      // Realm Cached Objects
-      cachedSurvey: cachedSurvey,
-      // Data
-      id: cachedSurvey.id,
-      status: cachedSurvey.status.length === 0 ? 'pending' : cachedSurvey.status, // pending, accepted, declined
-      user: cachedSurvey.user,
-      description: cachedSurvey.description,
-      isLoading: true
+      status: InvitationStatus.PENDING,
     }
   },
 
   componentWillMount() {
-    let cachedForms = loadCachedForms(this.state.id)
-    if (cachedForms.length === 0) {
-      this.fetchForms()
-    } else {
-      this.setSurveyDetails(cachedForms)
-    }
   },
 
   /* Methods */
-  fetchForms() {
-    let self = this
-    getSurveyForms(this.state.id, function(err, forms) {
-      self.setSurveyDetails(forms)
-    })
-  },
-
-
-  setSurveyDetails(forms){
-    let self = this
-    this.questionCount(forms, function(questionCount){
-      if (!forms[0]) {
-        alert('There are currently no forms available for this survey.')
-      } else if (questionCount === 0) {
-        alert('Error: Unable to fetch the Questions associated with this Survey.')
-      }
-      self.setState({
-        isLoading: false,
-        formCount: forms.length || 0,
-        questionCount: questionCount || 0
-       })
-    })
-  },
-
-  loadQuestions(form, callback) {
-    loadQuestions(form, function(err, questions){
-      if (questions) callback(questions)
-    })
-  },
-
-  questionCount(forms, callback){
-    let questionCount = 0,
-        self = this,
-        formCount = forms.length
-    forms.forEach(function(form, i){
-      cachedQuestions = loadCachedQuestions(form.id)
-      if (cachedQuestions.length === 0) {
-        self.loadQuestions(form, function(questions) {
-          questionCount += questions.length
-          if (formCount == i+1) callback(questionCount)
-        })
-      } else {
-        questionCount += loadCachedQuestions(form.id).length
-        callback(questionCount)
-      }
-    })
-  },
 
   acceptSurvey() {
-    let survey = this.state.cachedSurvey
-    this.setState({status: 'accepted'})
-    realm.write(() => {
-      survey.status = 'accepted'
-    })
-    checkSurveyTimeTriggers(survey, true)
-    this.selectForm()
+    const status = InvitationStatus.ACCEPTED;
+    this.setState({status: status}, () => {
+
+    });
+    markInvitationStatus(this.props.survey.id, status, (err, res) => {
+      if (err) {
```

Comment:
You do raise a good point. There is a chore in pivotal to re-send dirty submissions/invitations.  However, prior to this implementation I think the method `loadInvitations` should be modified to ignore updating dirty invitations with what was on the server.